File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/worker/constraints/SyncWorkerConstraints.java
package com.lasthopesoftware.bluewater.client.stored.worker.constraints;

import android.content.SharedPreferences;
import androidx.work.Constraints;
import androidx.work.NetworkType;
import com.lasthopesoftware.bluewater.ApplicationConstants;

public class SyncWorkerConstraints implements ConstrainSyncWork {

	private final SharedPreferences sharedPreferences;

	public SyncWorkerConstraints(SharedPreferences sharedPreferences) {
		this.sharedPreferences = sharedPreferences;
	}

	@Override
	public Constraints getCurrentConstraints() {
		final boolean isSyncOnWifiOnly = sharedPreferences.getBoolean(ApplicationConstants.PreferenceConstants.isSyncOnWifiOnlyKey, false);
		final Constraints.Builder builder = new Constraints.Builder();
		return builder
			.setRequiredNetworkType(isSyncOnWifiOnly ? NetworkType.UNMETERED : NetworkType.CONNECTED)
			.setRequiresCharging(sharedPreferences.getBoolean(ApplicationConstants.PreferenceConstants.isSyncOnPowerOnlyKey, false))
			.build();
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/scheduling/constraints/SyncWorkerConstraints.kt
package com.lasthopesoftware.bluewater.client.stored.scheduling.constraints

import androidx.work.Constraints
import androidx.work.NetworkType
import com.lasthopesoftware.bluewater.settings.repository.access.HoldApplicationSettings
import com.namehillsoftware.handoff.promises.Promise

class SyncWorkerConstraints(private val applicationSettings: HoldApplicationSettings) : ConstrainSyncWork {
	override val currentConstraints: Promise<Constraints>
		get() = applicationSettings.promiseApplicationSettings()
			.then { a ->
				Constraints.Builder()
					.setRequiredNetworkType(if (a.isSyncOnWifiOnly) NetworkType.UNMETERED else NetworkType.CONNECTED)
					.setRequiresCharging(a.isSyncOnPowerOnly)
					.build()
			}
}
